apache-rat-core/src/main/java/org/apache/rat/document/impl/ArchiveEntryDocument.java

```diff
@@ -37,44 +36,44 @@ public class ArchiveEntryDocument implements Document {
 
     private final MetaData metaData = new MetaData();
 
-    public ArchiveEntryDocument(File file, byte[] contents) throws RatException {
+    public ArchiveEntryDocument(File file, byte[] contents) {
         super();
         name = DocumentImplUtils.toName(file);
         this.contents = contents;
     }
 
+    @Override
     public MetaData getMetaData() {
         return metaData;
     }
 
+    @Override
     public String getName() {
         return name;
     }
 
+    @Override
     public InputStream inputStream() throws IOException {
         return new ByteArrayInputStream(contents);
     }
 
+    @Override
     public boolean isComposite() {
         return DocumentImplUtils.isZipStream(new ByteArrayInputStream(contents));
```

Comment:
Could we change the interface to internally wrap/add the ByteArrayInputStream in isZipStream() and autoclose it with a tryWithresources construct?